uPortal-core/src/main/java/org/apereo/portal/portlet/om/IPortletDefinition.java

```diff
@@ -59,8 +60,31 @@
      */
     boolean setPortletPreferences(List<IPortletPreference> portletPreferences);
 
+    /**
+     * The current lifecycle state is equal to the last entry in the lifecycle history that is not
+     * in the future.
+     */
     PortletLifecycleState getLifecycleState();
 
+    /**
+     * Adds the specified state to the lifecycle history at the present moment, removing any
+     * lifecycle changes scheduled for the future.
+     */
+    void setLifecycleState(PortletLifecycleState lifecycleState, IPerson user);
+
+    /**
+     * Schedules the specified state change for the specified moment, removing any lifecycle
+     * changes scheduled beyond that.
+     */
+    void setLifecycleState(PortletLifecycleState lifecycleState, IPerson user, Date timestamp);
```

Comment:
Indeed. It's not a simple JavaBean mutator, so avoiding `setX()` naming convention is a good call.